lib/api/core/plugins/pluginContext.js

```diff
@@ -252,4 +262,50 @@ function instantiateRequest(request, data, options = {}) {
   return target;
 }
 
+/**
+ * Returns a currified function of pluginsManager.registerStrategy
+ *
+ * @param  {Kuzzle} kuzzle 
+ * @param  {string} pluginName 
+ * @return {function} function taking a strategy name and properties, 
+ *                    registering it into kuzzle, and returning
+ *                    a promise
+ */
+function curryAddStrategy(kuzzle, pluginName) {
+  return function addStrategy(name, strategy) {
+    return new Bluebird((resolve, reject) => {
+      try {
+        kuzzle.pluginsManager.registerStrategy(pluginName, name, strategy);
+        resolve();
+      }
+      catch(err) {
+        reject(err);
+      }
+    });
+  };
+}
+
+/**
+ * Returns a currified function of pluginsManager.unregisterStrategy
+ *
+ * @param  {Kuzzle} kuzzle 
+ * @param  {string} pluginName 
+ * @return {function} function taking a strategy name and properties, 
+ *                    registering it into kuzzle, and returning
+ *                    a promise
+ */
+function curryRemoveStrategy(kuzzle, pluginName) {
+  return function removeStrategy(name) {
+    return new Bluebird((resolve, reject) => {
+      try {
+        kuzzle.pluginsManager.unregisterStrategy(pluginName, name);
```

Comment:
suggestion: add a hook `strategy:remove` to enable cluster plugin to propagate the action (see commente [here](https://github.com/kuzzleio/documentation/pull/339#discussion_r130080996)